### ISSUE TITLE
sitewide 508-defect-2 [COGNITION, SCREEN READER, MOBILE]: Mobile and screen reader users SHOULD know there is relevant info after the call-to-action button

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#16182]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** _Jennifer_

## User Story or Problem Statement

As a screen reader user, I need to know that there is additional relevant information after the call-to-action. Since there is a link to Sign in or create an account, I may just select that button and never know that there is information that clarifies the form, process, next steps, and how to get help completing the form.

## Details

**This is an issue across the site.** 

While this is a sitewide issue that should be addressed globally, the team should address this within this work. The Healthcare Questionnaire team also identified UX issues for sighted users with this pattern, especially on mobile.

The placement of the call-to-action (CTA) to sign in to use the service is at the top of the page. For sighted users on desktop it is clear there is additional information available. For screen reader users, there is no indication there is additional information that could be helpful. It is possible to add `aria-describedby` to the button to associate it with the following information. 

However, a larger issue is that this information is also not readily apparent for mobile users. It may be useful to consider removing the CTA from the top, ensuring users go through all the educational material before they use the CTA at the bottom of the page. This may require adopting a show/hide pattern for the subway steps, [as gov.uk does](https://www.gov.uk/apply-online-for-student-finance).

## Acceptance Criteria

- [ ] (Long term solution) Consider if it is desirable, viable, and feasible to remove the top CTA, potentially looking into research to see if this is desirable for users (esp mobile and screen reader), viable & feasible check with VSP (Trevor is aware).
- [ ] (Temporary remediation) The Sign in or create an account button has an `aria-describedby` associated with a unique id on the relevant following information.

**Update:** @AngelaFowler82 and I confirmed that `aria-describedby` will not read out for screen reader users with advanced verbosity settings. We've updated the AC to prioritize to remove the top CTA in favor of placing the CTA after relevant/important content.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: this project is in staging; the issue exists in production
* Devices: mobile and screen readers especially impacted

## Steps to Recreate

1. Enter `<https://staging.va.gov/decision-reviews/higher-level-review/request-higher-level-review-form-20-0996/introduction>` in browser
2. Start screenreading device listed in Environment
3. Start to read the page, without looking at it
4. Verify that there is no indication of the following support material
5. Optional: enter the url above on a mobile device
6. Verify that the CTA takes up the entire screen, and there is no indication of the following support content

#### Current

```html
<button class="va-button-primary">Sign in or create an account</button>
```

#### Recommendation

```html
<button aria-describedby="unique_id" class="va-button-primary">Sign in or create an account</button>

<div id="unique_id"… <!-- this could be on a container div or the h2 -->
```

### Option 2 - explore removing the top CTA

This is something that should be discussed with VSP, cc @1Copenut @sshein. On VSA we can:

1. Discuss the problem found
2. Explore and document actions
3. Document a solution that works for VSA teams
4. Propose to VSP design system council for review

Note: I believe this issue is also being discussed on the Healthcare work, and it may be helpful to connect with Kristen McConnell and myself. It could be helpful is VSA teams had a collaboration session to review this flow and brainstorm potential fixes. For example, Have the landing page only be the "Sign in or create an account", and then once signed-in, provide the subway steps task list before they begin the process. Others may have better ideas.

## WCAG or Vendor Guidance (optional)

* [WCAG 2.1 Level A - 1.3.2 Meaningful Sequence](https://www.wuhcag.com/meaningful-sequence/) - Users who rely on assistive technology (such as a screen reader) to interpret content, require content to be presented in a meaningful order. If this is presented out of sequence, users may become disorientated and will not understand the content.

## Screenshots or Trace Logs

![Screen Shot 2020-11-13 at 5 15 18 PM](https://user-images.githubusercontent.com/57469/99128924-25128100-25da-11eb-994f-38d0f5aa1fb2.png)

**Comment by 1Copenut** [comment]
@joshkimux Noah and I discussed this one a bit more in staging review. What do you think about adding a link above the call to action button that links to the heading/div you outlined? We're hesitant to link to too much additional information in a call to action because the screen reader user may not anticipate hearing such a long block read back when the button is under keyboard focus.

**Comment by Mottie** [comment]
I was just looking at this... it points to the `h2` in this case, so it should be okay?

Also, there is a second `Sign in or create an account` button at the bottom of the page.. I guess it's okay if both point to the same id?

**Comment by meganhkelley** [comment]
Passing to design system team! 👍 @caw310 @cvalarida

**Comment by Mottie** [comment]
sorry about that!

**Comment by joshkimux** [comment]
Roping @AngelaFowler82 into this. I'm sure she'll have great feedback to provide on this as a screen reader user. Looking back at the original recommendation, I want to test to see if `aria-describedby` will perform in the way we intended.

**Comment by joshkimux** [comment]
@Mottie update here. We confirmed that `aria-describedby` actually won't announce to advanced screen reader users who may have advanced verbosity settings that reduce how much is read out; `aria-describedby` specifically won't be read out, so they'll still face the potential issue of clicking a button without reading ahead.

That all being said, I'd like to propose that we should prioritize placing those call to actions lower in the page after important content (the second acceptance criteria). `aria-describedby` will be at best a temporary bandaid for this larger content/IA issue.

**Comment by Mottie** [comment]
> I'd like to propose that we should prioritize placing those call to actions lower in the page after important content (the second acceptance criteria). `aria-describedby` will be at best a temporary bandaid for this larger content/IA issue.

Like what's been done in the health care app? https://www.va.gov/health-care/apply/application/introduction

**Comment by joshkimux** [comment]
@Mottie yes, this exactly 💯

**Comment by bkjohnson** [comment]
Is this still an issue for the design system team to address?

**Comment by joshkimux** [comment]
@bkjohnson [According to the eligibility hierarchy on the design system], it looks like guidance aligns with what was proposed in this ticket where the sign in and call to action buttons will be placed below the subway map. @GnatalieH would you be able to confirm? If so, we can close this ticket out I think 🤞

**Comment by GnatalieH** [comment]
> @bkjohnson [According to the eligibility hierarchy on the design system](https://design.va.gov/patterns/benefit-applications#eligibility-hierarchy), it looks like guidance aligns with what was proposed in this ticket where the sign in and call to action buttons will be placed below the subway map. @GnatalieH would you be able to confirm? If so, we can close this ticket out I think 🤞

Well, according the design system entry, it's still OK to include FAQs and related content as optional elements after the primary call to action. The documentation doesn't call-out a Subway map specifically, however it does state critically important information should always be placed above the primary call to action. Does this work? I'm happy to add in guidelines that are specific to accessibility considerations if you'd like.

**Comment by joshkimux** [comment]
That should do the trick Natalie! I'd also love to sync up with you later about writing in guidance around a11y considerations specific to this ticket (more so to provide a _why_ to folks as to why the primary call to action should go below important information).

**Comment by GnatalieH** [comment]
@joshkimux sent ya a meeting invite just now to discuss next week. Thanks!
